src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -821,7 +831,7 @@ void QgsLayoutItemLegend::onAtlasEnded()
   updateFilterByMap();
 }
 
-QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
+QgsExpressionContext QgsLayoutItemLegend::createExpressionContext( bool replace ) const
```

Comment:
This method should not modify the internals.
If it's absolutely necessary, a new method `updateExpressionContext()` which is not const should be added.

Reply:
Since mExpContext is a new internal, and that the expression context is also a new method I thought I should bundle them together.

I have changed things and separated them.